### ISSUE TITLE
Demonology Warlock - Summon Demonic Tyrant not showing as a Cooldown or in Rotation

### ISSUE THREAD

**Issue by tommyruin** [Hekili/hekili#1104]
Hi - as title states, the hekili rotation isnt showing when to use the Demonic Tyrant as either a cooldown or as part of the rotation. 

build: v9.0.5-1.0.8
level: 60 (60)
class: WARLOCK
spec: demonology

talents: burning_rush
    darkfury
    demonic_calling
    demonic_consumption
    demonic_strength
    grimoire_felguard
    summon_vilefiend

pvptalents: none

covenant: necrolord

conduits: borne_of_blood = 5
   demonic_momentum = 6
   diabolic_bloodstone = 6
   tyrants_soul = 6

soulbinds: [plague_deviser_marileth]
   oozs_frictionless_coating = 1
   plagueborn_cleansing_slime = 1
   ultimate_form = 1
   volatile_solvent = 1

sets: none

gear: cinch_of_infinite_tightness = 1
    confidants_favored_cap = 1
    courtiers_costume_trousers = 1
    death_gods_signet = 1
    drape_of_twisted_loyalties = 1
    entwined_gorger_tendril = 1
    fleshscourge_sickle = 1
    gray_woolen_shirt = 1
    grimveiled_bracers = 1
    impossibly_oversized_mitts = 1
    infinitely_divisible_ooze = 1
    malevolent_invokers_crystal = 1
    reanimators_mantle = 1
    robes_of_the_cursed_commando = 1
    sin_stained_pendant = 1
    soulletting_ruby = 1
    windscale_moccasins = 1

legendaries: wilfreds_sigil_of_superior_summoning = 1

itemIDs: 173249, 178740, 178769, 178809, 178827, 178848, 179322, 179355, 180123, 182998, 183011, 183021, 183022, 183028, 184258, 184266, 2587

settings: aoe = 3
    buffPadding = 0
    custom1Name = Custom 1
    custom2Name = Custom 2
    cycle = true
    cycle_min = 6
    damage = true
    damageDots = false
    damageExpiration = 6
    damagePets = false
    damageRange = 0
    debuffPadding = 0
    enabled = true
    enhancedRecheck = false
    gcdSync = true
    maxRefresh = 10
    maxTime = 33
    nameplateRange = 8
    nameplates = false
    package = SimC Demo
    petbased = true
    potion = spectral_intellect
    potionsReset = 20180919.1
    throttleRefresh = false
    throttleTime = false
    implosion_imps = 6

toggles: cooldowns = true
    custom1 = false 
    custom2 = false 
    defensives = false 
    essences = true 
    interrupts = false 
    mode = single 
    potions = false

**Comment by Hekili** [comment]
You'll need to follow the ticket instructions and include the snapshot.

**Comment by Hekili** [comment]
It also looks like you did your own SimC import, since your priority name is not one of the defaults.  If you import your own priority, you have to make it work on your own as well.

**Comment by tommyruin** [comment]
Hi Hekili - apologies if I did things incorrectly. I have switched to the Demonology SimC profile to make things easier (a similar problem is occuring though). 

Does this snapshot make sense?

Click here and press CTRL+A, CTRL+C to copy the snapshot.

build: v9.0.5-1.0.8
level: 60 (60)
class: WARLOCK
spec: demonology

talents: burning_rush
    darkfury
    demonic_calling
    demonic_consumption
    demonic_strength
    grimoire_felguard
    summon_vilefiend

pvptalents: none

covenant: necrolord

conduits: borne_of_blood = 5
   demonic_momentum = 6
   diabolic_bloodstone = 6
   tyrants_soul = 6

soulbinds: [plague_deviser_marileth]
   oozs_frictionless_coating = 1
   plagueborn_cleansing_slime = 1
   ultimate_form = 1
   volatile_solvent = 1

sets: none

gear: cinch_of_infinite_tightness = 1
    confidants_favored_cap = 1
    courtiers_costume_trousers = 1
    death_gods_signet = 1
    drape_of_twisted_loyalties = 1
    entwined_gorger_tendril = 1
    fleshscourge_sickle = 1
    gray_woolen_shirt = 1
    grimveiled_bracers = 1
    impossibly_oversized_mitts = 1
    infinitely_divisible_ooze = 1
    malevolent_invokers_crystal = 1
    reanimators_mantle = 1
    robes_of_the_cursed_commando = 1
    sin_stained_pendant = 1
    soulletting_ruby = 1
    windscale_moccasins = 1

legendaries: wilfreds_sigil_of_superior_summoning = 1

itemIDs: 173249, 178740, 178769, 178809, 178827, 178848, 179322, 179355, 180123, 182998, 183011, 183021, 183022, 183028, 184258, 184266, 2587

settings: aoe = 3
    buffPadding = 0
    custom1Name = Custom 1
    custom2Name = Custom 2
    cycle = true
    cycle_min = 6
    damage = true
    damageDots = false
    damageExpiration = 6
    damagePets = false
    damageRange = 0
    debuffPadding = 0
    enabled = true
    enhancedRecheck = false
    gcdSync = true
    maxRefresh = 10
    maxTime = 33
    nameplateRange = 8
    nameplates = false
    package = Demonology SimC
    petbased = true
    potion = spectral_intellect
    potionsReset = 20180919.1
    throttleRefresh = false
    throttleTime = false
    implosion_imps = 6

toggles: cooldowns = true
    custom1 = false 
    custom2 = false 
    defensives = false 
    essences = true 
    interrupts = false 
    mode = single 
    potions = false 

targets:
    dmg          -  0 - Creature-0-4247-2222-13648-174484-0000AA06C7


player_buffs:
   186406 - sign_of_the_critter                      -   1 - 3600.00
   311452 - vantus_rune_stone_legion_generals        -   1 - 213626.52
   324867 - fleshcraft                               -   1 - 59.06
   108366 - soul_leech                               -   1 - 14.18
   205146 - demonic_calling                          -   1 - 17.45
   264173 - demonic_core                             -   1 - 17.35

player_debuffs:

target_buffs:

target_debuffs:
    30213 - legion_strike                            -   1 - 5.08


New Recommendations for [ Cooldowns ] requested at 20:39:36 ( 645776.24 ); using built-in ( Demonology SimC ) priority.
Is Tyrant Ready?: Yes
 - Dreadstalkers: 0, 0.00
 - Vilefiend    : 1, 1.76
 - Grim Felguard: 1, 0.76
 - Wild Imps    : 5, 8.02
Next Demon Exp. : 0.76

RECOMMENDATION #1 ( Offset: 0.00, GCD: 0.00, Casting: 0.00 ).

    Resources: mana[ 50000.00 / 50000.00 ], soul_shards[ 0.00 / 5.00 ]
    Checking if I'm casting ( No ) and if it is a channel ( No ).
    No whitelist.
    
    Processing precombat action list [ Demonology SimC - precombat ].
        
        Current recommendation was NO ACTION at +60.00s.
        
        1.   fel_domination ( precombat - 1 ) - ability disabled ( display )
        Time spent on this action:  0.03ms
        TimeData:Demonology SimC-precombat-1:fel_domination:0.03
        
        2.   summon_felguard ( precombat - 2 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-precombat-2:summon_felguard:0.02
        
        3.   power_siphon ( precombat - 3 ) - talent [ power_siphon ] missing
        Time spent on this action:  0.03ms
        TimeData:Demonology SimC-precombat-3:power_siphon:0.03
        
        4.   demonbolt ( precombat - 4 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-precombat-4:demonbolt:0.02
    
    Completed precombat action list [ Demonology SimC - precombat ].
    
    Processing default action list [ Demonology SimC - default ].
        
        Current recommendation was NO ACTION at +60.00s.
        
        5.   spell_lock ( default - 1 ) - ability unknown
        Time spent on this action:  0.03ms
        TimeData:Demonology SimC-default-1:spell_lock:0.03
        
        6.   call_felhunter ( default - 2 ) - PvP talent [ call_felhunter ] missing
        Time spent on this action:  0.08ms
        TimeData:Demonology SimC-default-2:call_felhunter:0.08
        
        7.   axe_toss ( default - 3 ) - ability disabled ( display )
        Time spent on this action:  0.03ms
        TimeData:Demonology SimC-default-3:axe_toss:0.03
        
        8.   devour_magic ( default - 4 ) - ability unknown
        Time spent on this action:  0.03ms
        TimeData:Demonology SimC-default-4:devour_magic:0.03
        
        9.   call_action_list ( default - 5 )
        There is no criteria for off_gcd.
        Action list (off_gcd) was found.
        + off_gcd
        Current recommendation was NO ACTION at +60.00s.
        
        10.  berserking ( off_gcd - 1 ) - ability unknown
        Time spent on this action:  0.03ms
        TimeData:Demonology SimC-off_gcd-1:berserking:0.03
        
        11.  potion ( off_gcd - 2 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-off_gcd-2:potion:0.02
        
        12.  blood_fury ( off_gcd - 3 ) - ability unknown
        Time spent on this action:  0.03ms
        TimeData:Demonology SimC-off_gcd-3:blood_fury:0.03
        
        13.  fireblood ( off_gcd - 4 ) - ability unknown
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-off_gcd-4:fireblood:0.02
        Returned from list (off_gcd), current recommendation is NO ACTION (+60.00).
        - off_gcd
        
        14.  run_action_list ( default - 6 )
        Criteria for tyrant_prep FAIL at +0.00 - cooldown.summon_demonic_tyrant.remains[0.00] < 4 & ! tyrant_ready[true]
        
        15.  run_action_list ( default - 7 )
        Criteria for summon_tyrant PASS at +0.00 - tyrant_ready[true]
        Action list (summon_tyrant) was found.
        + [summon_tyrant]
        Current recommendation was NO ACTION at +60.00s.
        
        16.  hand_of_guldan ( summon_tyrant - 1 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-summon_tyrant-1:hand_of_guldan:0.02
        
        17.  demonbolt ( summon_tyrant - 2 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-summon_tyrant-2:demonbolt:0.02
        
        18.  shadow_bolt ( summon_tyrant - 3 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-summon_tyrant-3:shadow_bolt:0.02
        
        19.  grimoire_felguard ( summon_tyrant - 4 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-summon_tyrant-4:grimoire_felguard:0.02
        
        20.  summon_vilefiend ( summon_tyrant - 5 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-summon_tyrant-5:summon_vilefiend:0.02
        
        21.  call_dreadstalkers ( summon_tyrant - 6 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-summon_tyrant-6:call_dreadstalkers:0.02
        
        22.  hand_of_guldan ( summon_tyrant - 7 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-summon_tyrant-7:hand_of_guldan:0.02
        
        23.  demonbolt ( summon_tyrant - 8 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-summon_tyrant-8:demonbolt:0.02
        
        24.  soul_strike ( summon_tyrant - 9 ) - ability unknown
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-summon_tyrant-9:soul_strike:0.02
        
        25.  shadow_bolt ( summon_tyrant - 10 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-summon_tyrant-10:shadow_bolt:0.02
        
        26.  summon_demonic_tyrant ( summon_tyrant - 11 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-summon_tyrant-11:summon_demonic_tyrant:0.02
        
        27.  shadow_bolt ( summon_tyrant - 12 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-summon_tyrant-12:shadow_bolt:0.02
        Returned from list (summon_tyrant), current recommendation is NO ACTION (+60.00).
        - [summon_tyrant]
        Added summon_tyrant to blocklist as it was called via RAL.
        
        28.  summon_vilefiend ( default - 8 ) - ability disabled ( display )
        Time spent on this action:  0.01ms
        TimeData:Demonology SimC-default-8:summon_vilefiend:0.01
        
        29.  call_dreadstalkers ( default - 9 ) - ability disabled ( display )
        Time spent on this action:  0.01ms
        TimeData:Demonology SimC-default-9:call_dreadstalkers:0.01
        
        30.  doom ( default - 10 ) - talent [ doom ] missing
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-default-10:doom:0.02
        
        31.  demonic_strength ( default - 11 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-default-11:demonic_strength:0.02
        
        32.  bilescourge_bombers ( default - 12 ) - talent [ bilescourge_bombers ] missing
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-default-12:bilescourge_bombers:0.02
        
        33.  implosion ( default - 13 ) - ability disabled ( display )
        Time spent on this action:  0.03ms
        TimeData:Demonology SimC-default-13:implosion:0.03
        
        34.  implosion ( default - 14 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-default-14:implosion:0.02
        
        35.  hand_of_guldan ( default - 15 ) - ability disabled ( display )
        Time spent on this action:  0.01ms
        TimeData:Demonology SimC-default-15:hand_of_guldan:0.01
        
        36.  hand_of_guldan ( default - 16 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-default-16:hand_of_guldan:0.02
        
        37.  call_action_list ( default - 17 )
        Criteria for covenant PASS at +0.00 - ( covenant.necrolord[true] | covenant.night_fae[false] ) & ! talent.nether_portal.enabled[false]
        Action list (covenant) was found.
        + covenant
        Current recommendation was NO ACTION at +60.00s.
        
        38.  impending_catastrophe ( covenant - 1 ) - ability unknown
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-covenant-1:impending_catastrophe:0.02
        
        39.  scouring_tithe ( covenant - 2 ) - ability unknown
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-covenant-2:scouring_tithe:0.02
        
        40.  scouring_tithe ( covenant - 3 ) - ability unknown
        Time spent on this action:  0.01ms
        TimeData:Demonology SimC-covenant-3:scouring_tithe:0.01
        
        41.  soul_rot ( covenant - 4 ) - ability unknown
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-covenant-4:soul_rot:0.02
        
        42.  decimating_bolt ( covenant - 5 ) - ability disabled ( display )
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-covenant-5:decimating_bolt:0.02
        Returned from list (covenant), current recommendation is NO ACTION (+60.00).
        - covenant
        
        43.  demonbolt ( default - 18 ) - ability disabled ( display )
        Time spent on this action:  0.01ms
        TimeData:Demonology SimC-default-18:demonbolt:0.01
        
        44.  grimoire_felguard ( default - 19 ) - ability disabled ( display )
        Time spent on this action:  0.01ms
        TimeData:Demonology SimC-default-19:grimoire_felguard:0.01
        
        45.  use_items ( default - 20 )
        There is no criteria for Use Items.
        + items
        Current recommendation was NO ACTION at +60.00s.
        
        46.  soulletting_ruby ( items - 1 )
        The action (soulletting_ruby) is usable at (0.00 + 0.00).
         - the action is ready before the current recommendation (at +0.00 vs. +60.00).
        Blocking list ( summon_tyrant ) called from ( Demonology SimC:default:7 ) would BLOCK at 0.00.
        tyrant_ready[true]
         - this entry would not be reached at the current time via the current action list path (0.00).
        There were no recheck events to check.
        Time spent on this action:  0.26ms
        TimeData:UseItems-items-1:soulletting_ruby:0.26:Post-TTR and Essential:0.10:Post Cycle:0.00:Post Usable:0.03:Post Ready/Clash:0.01:Post Stack:0.03:Pre-Recheck:0.01:Post-Recheck Times:0.01:Post Recheck:0.01
        Returned from Use Items; current recommendation is NO ACTION (+60.00).
        - items
        
        47.  power_siphon ( default - 21 ) - talent [ power_siphon ] missing
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-default-21:power_siphon:0.02
        
        48.  soul_strike ( default - 22 ) - ability unknown
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-default-22:soul_strike:0.02
        
        49.  call_action_list ( default - 23 )
        There is no criteria for covenant.
        Action list (covenant) was found.
        + covenant
        Current recommendation was NO ACTION at +60.00s.
        
        50.  impending_catastrophe ( covenant - 1 ) - ability unknown
        Time spent on this action:  0.01ms
        TimeData:Demonology SimC-covenant-1:impending_catastrophe:0.01
        
        51.  scouring_tithe ( covenant - 2 ) - ability unknown
        Time spent on this action:  0.01ms
        TimeData:Demonology SimC-covenant-2:scouring_tithe:0.01
        
        52.  scouring_tithe ( covenant - 3 ) - ability unknown
        Time spent on this action:  0.01ms
        TimeData:Demonology SimC-covenant-3:scouring_tithe:0.01
        
        53.  soul_rot ( covenant - 4 ) - ability unknown
        Time spent on this action:  0.02ms
        TimeData:Demonology SimC-covenant-4:soul_rot:0.02
        
        54.  decimating_bolt ( covenant - 5 ) - ability disabled ( display )
        Time spent on this action:  0.01ms
        TimeData:Demonology SimC-covenant-5:decimating_bolt:0.01
        Returned from list (covenant), current recommendation is NO ACTION (+60.00).
        - covenant
        
        55.  shadow_bolt ( default - 24 ) - ability disabled ( display )
        Time spent on this action:  0.01ms
        TimeData:Demonology SimC-default-24:shadow_bolt:0.01
    
    Completed default action list [ Demonology SimC - default ].
    Recommendation is NO ACTION at 0.00 + 60.00.
    Recommendation #1 is NO ACTION at 60.00s (60.00s).
    Time spent generating recommendations:  2.79ms
    
    The problem is that it is not recommending tyrant prior to Grimoire Felguard, Summon Vilefiend and Dreadstalkers dropping off. I think it lets Dreadstalkers drop off and then suggests Tyrant.

**Comment by Hekili** [comment]
You'll have to do your own editing if you want to import.  If you want to work on that, I'd suggest comparing the default SimC priority vs. what you see in the addon, and make the same kinds of changes to what you've imported.

**Comment by tommyruin** [comment]
Apologies. It seems like it was an issue with the profile I have imported. However, the logic for summon demonic tyrant does seem strange regardless and doesnt seem to maximise the damage from holding spells for Demonic consumption maximisation.

**Comment by LucianDK** [comment]
I am new to demonology warlocks and I have experienced a similar problem to the OP here. But I discovered why it happens.

The culprit is the legendary Wilfred's Sigil of Superior Summoning.
Prior to obtaining it I had the cooldowns showing in a seperate queue, wherein Summon Demonic Tyrant showed just fine. But after aquiring the legendary it does not do so anymore. Upon untoggling seperate cooldowns, Summon Demonic Tyrant shows in the primary queue again. Only by removing the legendary again can I get it to show in the seperate cooldown queue when toggled on.

[Showseperatelywithring.txt](https://github.com/Hekili/hekili/files/7376985/Showseperatelywithring.txt)
[showseperatelywithoutring.txt](https://github.com/Hekili/hekili/files/7376986/showseperatelywithoutring.txt)
[Showinprimarywithring.txt](https://github.com/Hekili/hekili/files/7376987/Showinprimarywithring.txt)

**Comment by Hekili** [comment]
Those snapshots don't support what you're saying.  Summon Demonic Tyrant isn't shown in the separate display at the time you took the snapshot because its criteria aren't met -- you only have 3 Wild Imps and none of your major demons (Felguard, Vilefiend) out.

**Comment by LucianDK** [comment]
Ok, let me try again:
Here is the snap from doing the rotation with the wilfred legendary ring and no seperate cooldown queue toggled. Where Demonic Tyrant appears in the primary queue as expected.
[WithRingAndNoSeperateCooldown.txt](https://github.com/Hekili/hekili/files/7378138/WithRingAndNoSeperateCooldown.txt)


Here is the rotation with the ring and seperate cooldown queue toggled. And now Demonic Tyrant never appears in the seperate cooldown queue:
[WithRingAndSeperateCooldown.txt](https://github.com/Hekili/hekili/files/7378141/WithRingAndSeperateCooldown.txt)

And finally an image of me having removed the ring, and Demonic Tyrant shows up in the seperate cooldown queue
![DemonicTyrantwithoutRing](https://user-images.githubusercontent.com/48681832/138028749-d2e8e72c-5146-4474-99f6-394507ba58af.png)

Having the ring on, it is impossible to get Demonic Tyrant to show in the seperate cooldown queue.

I hope this is more telling.

**Comment by marekCs** [comment]
I have a similar problem and since one thread is already open, I didn't start a new one. In short, I never get a recommendation for Summon Demonic Tyrant even when I have "charged". Talents the same, brick setup 9.1.5. I'm using Fixed dual display. Default Priory from Hekili.

**Comment by TheCobblestoner** [comment]
Samesies. Only recently got the addon, and customising my own looks daunting but base setting - Demonic Tyrant (additionally Grimoire: Felguard) never show up in the rotation. Like, EVER.
